src/Model/HasAttributesTrait.php

```diff
@@ -65,6 +67,14 @@ trait HasAttributesTrait
      */
     private $originalAttributes = [];
 
+    /**
+     * Attributes that are cast to another types when fetched from database.
+     *
+     * @var array
+     */
+    protected $casts = [];
```

Comment:
What about set type?
```suggestion
    protected array $casts = [];
```

Reply:
Done: https://github.com/leroy-merlin-br/mongolid/pull/220/commits/ea1efb32835be3a0deba98888a58b39cbd8409d8